src/close-matches.js

```diff
@@ -0,0 +1,86 @@
+import {makeNormalizer} from './matches'
+
+const initializeDpTable = (rows, columns) => {
+  const dp = Array(rows + 1)
+    .fill()
+    .map(() => Array(columns + 1).fill())
+
+  // fill rows
+  for (let i = 0; i <= rows; i++) {
+    dp[i][0] = i
+  }
+
+  // fill columns
+  for (let i = 0; i <= columns; i++) {
+    dp[0][i] = i
+  }
+  return dp
+}
+
+export const calculateLevenshteinDistance = (text1, text2) => {
```

Comment:
Rather than maintain/test this ourselves, I'd much rather use a third party module. I've used https://github.com/sindresorhus/leven before and it works well.